### PR TITLE
[Docs] Recommend Ray Jobs in client docs

### PR DESCRIPTION
**Pull request by ckw017**
## Why are these changes needed?

Ray jobs are the recommended way of starting remote workloads.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/running-applications/job-submission/ray-client.rst [hunk 1]
```diff
@@ -26,13 +26,24 @@ By changing ``ray.init()`` to ``ray.init("ray://<head_node_host>:<port>")``, you
    do_work.remote(2)
    #....
 
+When to use Ray Client
+----------------------
+
+Ray Client should be used when you want to connect a script or an interactive shell session to a **remote** cluster.
+
+* Use ``ray.init("ray://<head_node_host>:10001")`` (Ray Client) if you've set up a remote cluster at ``<head_node_host>`` and you want to do interactive work. This will connect your local script or shell to the cluster. See the section on :ref:`using Ray Client<how-do-you-use-the-ray-client>` for more details on setting up your cluster.
+* Use ``ray.init("localhost:<port>")`` (non-client connection, local address) if you're developing locally or on the head node of your cluster and you have already started the cluster (i.e. ``ray start --head`` has already been run)
```
Comment (DmitriGekhtman) [inline doc/source/cluster/running-applications/job-submission/ray-client.rst:35]: Actually, as of Ray 2.0.0, `ray.init()` is now the way to go if you've run `ray start` already.
You only need to specify an address if there are multiple Ray instances running locally.

**Review by DmitriGekhtman — APPROVED** [review]

File: doc/source/cluster/running-applications/job-submission/ray-client.rst [hunk 2]
```diff
@@ -26,13 +26,23 @@ By changing ``ray.init()`` to ``ray.init("ray://<head_node_host>:<port>")``, you
...
+
+Ray Client should be used when you want to connect a script or an interactive shell session to a **remote** cluster.
+
+* Use ``ray.init("ray://<head_node_host>:10001")`` (Ray Client) if you've set up a remote cluster at ``<head_node_host>`` and you want to do interactive work. This will connect your local script or shell to the cluster. See the section on :ref:`using Ray Client<how-do-you-use-the-ray-client>` for more details on setting up your cluster.
+* Use ``ray.init()`` (non-client connection, no address specified) if you're developing locally and want to connect to an existing cluster (i.e. ``ray start --head`` has already been run), or automatically create a local cluster and attach directly to it. This can also be used for :ref:`Ray Job <jobs-overview>` submission.
+
+Ray Client is useful for developing interactively in a local Python shell. However, it requires a stable connection to the remote cluster and will terminate the workload if the connection is lost for :ref:`more than 30 seconds <client-disconnections>`. If you have a long running workload that you want to run on your cluster, we recommend using :ref:`Ray Jobs <jobs-overview>` instead.
```
Comment (jjyao) [inline doc/source/cluster/running-applications/job-submission/ray-client.rst:37]: Besides a stable connection. Another common issue is inconsistent local and remote environment.

For developing interactively, should we recommend like remote vs code or ssh?

cc @stephanie-wang

Comment (DmitriGekhtman) [reply doc/source/cluster/running-applications/job-submission/ray-client.rst:37]: cc @richardliaw @anabranch

Comment (DmitriGekhtman) [reply doc/source/cluster/running-applications/job-submission/ray-client.rst:37]: It's a good question -- but we don't need to figure out the golden path for interactive development in this PR :)

File: doc/source/cluster/running-applications/job-submission/ray-client.rst [hunk 3]
```diff
@@ -26,13 +26,23 @@ By changing ``ray.init()`` to ``ray.init("ray://<head_node_host>:<port>")``, you
    do_work.remote(2)
    #....
 
+When to use Ray Client
+----------------------
+
+Ray Client should be used when you want to connect a script or an interactive shell session to a **remote** cluster.
```
Comment (wuisawesome) [inline doc/source/cluster/running-applications/job-submission/ray-client.rst:32]: Should we just say interactively develop on a remote cluster? Specifically you shouldn't use it to "fire and forget" a script on a remote cluster (you should just submit it as a job instead)

Comment (ckw017) [reply doc/source/cluster/running-applications/job-submission/ray-client.rst]: Removed mentions of running scripts with client

**Comment by pcmoritz** [comment]
Thanks for fixing this! I think we should still link the versioning issues in `When to use Ray Client` (i.e. link to the section in the doc further below), but we can do this in a follow up PR :)

**Pull request merged**